gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/NonAbstractGradleType.java

```diff
@@ -70,6 +106,55 @@ public Description matchClass(ClassTree tree, VisitorState state) {
         return Description.NO_MATCH;
     }
 
+    @Override
+    public Description matchMethod(MethodTree method, VisitorState state) {
+        ClassTree enclosingClass = ASTHelpers.findEnclosingNode(state.getPath(), ClassTree.class);
+        if (enclosingClass == null) {
+            return Description.NO_MATCH;
+        }
+
+        if (!IS_TASK.matches(enclosingClass, state)) {
+            return Description.NO_MATCH;
+        }
+
+        if (!isManagedPropertyGetter(method, state)) {
+            return Description.NO_MATCH;
+        }
+
+        if (!isAbstract(ASTHelpers.getSymbol(method))) {
+            return buildDescription(method)
+                    .setMessage(ABSTRACT_PROPERTY_METHOD_MSG)
+                    .build();
+        }
+        if (!method.getName().toString().startsWith("get")) {
+            return buildDescription(method)
+                    .setMessage(PROPERTY_GETTER_NAMING_MSG)
+                    .build();
+        }
```

Comment:
Might as well report both at once - the way to do this is give the description to `state.reportMatch` rather than returning, so you can report multiple issues.

Reply:
You can find subtypes by doing `state.getTypes().isSubType()`.

Reply:
Might be worth writing a check about this case just to be sure...